### PR TITLE
Lambda functions and Enum Homework

### PR DESCRIPTION
**Pull request by chindhur**

### CODE REVIEW

File: src/main/java/com/airport/Flight.java [hunk 1]
```diff
@@ -1,8 +1,12 @@
 package com.airport;
```
Comment (ankochergin) [inline src/main/java/com/airport/Flight.java:1]: check formatting in this file

File: src/main/java/com/utils/UniqueWords.java [hunk 2]
```diff
@@ -6,8 +6,38 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import java.util.List;
+
+interface IFindItem<T> {
+    // a method that calculates the number of occurrences
+    // of a given element in an array of numbers
+    int Search(T item, List<T> items);
```
Comment (ankochergin) [inline src/main/java/com/utils/UniqueWords.java]:
```suggestion
    int search(T item, List<T> items);
```

File: src/main/java/com/airport/FlightStatus.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    ON_TIME("On Time"),
+    DELAYED("Delayed"),
+    CANCELLED("Cancelled"),
+    BOARDING("Boarding"),
+    DEPARTED("Departed"),
+    LANDED("Landed");
+
+    private String displayName;
+
+    FlightStatus(String displayName) {
```
Comment (ankochergin) [inline src/main/java/com/airport/FlightStatus.java]:
```suggestion
    private FlightStatus(String displayName) {
```

Comment (ankochergin) [reply src/main/java/com/airport/FlightStatus.java]: same in other enums

File: src/main/java/com/utils/UniqueWords.java [hunk 4]
```diff
@@ -6,8 +6,38 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import java.util.List;
+
+interface IFindItem<T> {
+    // a method that calculates the number of occurrences
+    // of a given element in an array of numbers
+    int search(T item, List<T> items);
+}
```
Comment (ankochergin) [inline src/main/java/com/utils/UniqueWords.java]: it should be in s separate file

File: src/main/java/com/interfaces/IFilterDetails.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package com.interfaces;
+
+public interface IFilterDetails<T> {
+        boolean filter(T t);
+}
```
Comment (ankochergin) [inline src/main/java/com/interfaces/IFilterDetails.java]: formatting

File: src/main/java/com/interfaces/IPrintDetails.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package com.interfaces;
+
+@FunctionalInterface
+public interface IPrintDetails<T> {
+        void print(T t);
+}
```
Comment (ankochergin) [inline src/main/java/com/interfaces/IPrintDetails.java]: formatting

File: src/main/java/com/interfaces/IFindItem.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.interfaces;
+
+import java.util.List;
+
+public interface IFindItem<T> {
```
Comment (ankochergin) [inline src/main/java/com/interfaces/IFindItem.java:6]: @FunctionalInterface

File: src/main/java/com/people/Designation.java [hunk 8]
```diff
@@ -1,9 +1,39 @@
...
+        return this == AIRHOSTESS;
+    }
+
+    public boolean isCargoManager() {
+        return this == CARGO_MANAGER;
+    }
+
+    public boolean isCustomerCare() {
+        return this == CUSTOMERCARE;
+    }
```
Comment (ankochergin) [inline src/main/java/com/people/Designation.java]: I thing these methods are redundant

File: src/main/java/com/people/PersonChecker.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package com.people;
+
+public class PersonChecker {
+
```
Comment (ankochergin) [inline src/main/java/com/people/PersonChecker.java]: empty interface

File: src/main/java/com/airport/Flight.java [hunk 10]
```diff
@@ -1,45 +1,53 @@
...
+    protected String flightId;
+    protected String flightName;
     protected String source;
     protected String destination;
     protected Double price;
+    List<FoodMenu> menuList;
+    List<Passenger> checkedInPassengers;
+    //Implements Consumer Lambda- notifies boarding of a passenger
+    Consumer<Passenger> notifyBoarding =
+            passenger -> Printer.print(passenger.getName() + " boarded flight");
```
Comment (ankochergin) [inline src/main/java/com/airport/Flight.java]: what is this variable for?

File: src/main/java/com/airport/Flight.java [hunk 11]
```diff
@@ -1,45 +1,53 @@
...
+
+    static {
+        Printer.print("Have a Nice and Happy Journey with our Airlines");
+    }
+
     private final int NO_OF_WHEELS = 18;
+    protected String flightId;
+    protected String flightName;
     protected String source;
     protected String destination;
```
Comment (ankochergin) [inline src/main/java/com/airport/Flight.java]: static block should be after or before variables, not splitting them

File: src/main/java/com/airport/RunwayStatus.java [hunk 12]
```diff
@@ -1,8 +1,30 @@
...
+        return this == IDLE;
+    }
+
+    public boolean isTakeOff() {
+        return this == TAKEOFF;
+    }
+
+    public boolean isLanding() {
+        return this == LANDING;
+    }
```
Comment (ankochergin) [inline src/main/java/com/airport/RunwayStatus.java]: redundant methods

File: src/main/java/com/airport/Flight.java (hunk 10 above)
Comment (chindhur) [reply src/main/java/com/airport/Flight.java]: this notifies boarding of a passenger

**Pull request merged**